modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/ExecutionDistributionProviderImpl.java

```diff
@@ -205,4 +230,46 @@ private CompletableFuture<List<TokenizedAssignments>> allReplicas(
             });
         });
     }
+
+    static class DistributionHolderImpl implements DistributionHolder {
+        private final List<String> nodes;
+        private final Map<IgniteTable, List<TokenizedAssignments>> assignmentsPerTable;
+        private final Map<String, List<String>> nodesPerView;
+
+        DistributionHolderImpl(
+                List<String> nodes,
+                Map<IgniteTable, List<TokenizedAssignments>> assignmentsPerTable,
+                Map<String, List<String>> nodesPerView) {
+            this.nodes = nodes;
+            this.assignmentsPerTable = assignmentsPerTable;
+            this.nodesPerView = nodesPerView;
+        }
+
+        @Override
+        public List<String> nodes() {
+            return nodes;
+        }
+
+        @Override
+        public List<TokenizedAssignments> tableAssignments(IgniteTable table) {
+            return assignmentsPerTable.get(table);
+        }
+
+        @Override
+        public List<String> viewNodes(String viewName) {
+            return nodesPerView.get(viewName);
+        }
+    }
+
+    /** Nodes distribution information holder. */
+    public interface DistributionHolder {
+        /** Whole collection of participating nodes. */
+        List<String> nodes();
+
+        /** Nodes table assigned to. */
```

Comment:
it can return `null` as and the following method

Reply:
honestly, i don`t understand why and where `null` can be returned, can you clarify please ?

Reply:
this interface is no longer exist